### PR TITLE
packer: update the script to download & unpack Caddy

### PR DESCRIPTION
**Pull request by Mohammed90**
Update 20-caddy.sh to accommodate the version format of the asset & untar the asset

### CODE REVIEW

**Comment by Mohammed90** [comment]
What's the best way to handle verifying the checksum? Should I copy the corresponding checksum line into the script file?

File: packer/scripts/20-caddy.sh [hunk 1]
```diff
@@ -11,8 +12,9 @@ useradd --system \
 
 chown -R caddy:caddy /var/lib/caddy
 
-VERSION_SLUG=$(echo "$CADDY_VERSION" | sed 's/-/_/g' | sed 's/\.[0-9]*\.[0-9]*//g' | sed 's/v//g' | sed 's/\.//g')
-wget -qO /usr/bin/caddy https://github.com/caddyserver/caddy/releases/download/"${CADDY_VERSION}"/caddy"${VERSION_SLUG}"_linux_amd64
+VERSION_SLUG=${CADDY_VERSION//v/}
```
Comment (mholt) [inline packer/scripts/20-caddy.sh:15]: I have to admit I do not understand this line

File: packer/scripts/20-caddy.sh [hunk 2]
```diff
@@ -11,8 +12,9 @@ useradd --system \
...
+wget -qO /tmp/caddy.tar.gz https://github.com/caddyserver/caddy/releases/download/"${CADDY_VERSION}"/caddy_"${VERSION_SLUG}"_linux_amd64.tar.gz
+tar x -z -f /tmp/caddy.tar.gz -C /usr/bin caddy
```
Comment (mholt) [inline packer/scripts/20-caddy.sh:17]:
```suggestion
tar -xzf /tmp/caddy.tar.gz -C /usr/bin caddy
```

A little more consistent and readable? Just a nit :)

File: packer/scripts/20-caddy.sh (hunk 1 above)
Comment (Mohammed90) [reply packer/scripts/20-caddy.sh:15]: It is equivalent to this `echo $CADDY_VERSION | sed 's/v//g'`. I didn't know this syntax, but I use Shellcheck to lint the shell scripts and to warn of possibly risky lines, and it suggested this alternative to sed. I thought it was cool and works as intended.

**Pull request merged**